src/mesh/voxel_cloud.rs

```diff
@@ -228,6 +239,139 @@ impl VoxelCloud {
         }
     }
 
+    /// Computes boolean intersection (logical AND operation) of the current and
+    /// another Voxel cloud. The Voxel clouds have to intersect and the voxel
+    /// dimension must be equal. The current Voxel cloud will be mutated and
+    /// resized to the size and position of an intersection of the two Voxel
+    /// clouds (not the contained volumes).
+    pub fn boolean_intersection_mut(&mut self, other: &VoxelCloud) {
```

Comment:
This could be rewritten so that `other` is many `others`. The we can get rid of the immutable version of this function. On the downside, the caller has to take care of preparing a good "self" for such an operation. Specifically, working with mesh groups will require popping the first one and then exposing it to the rest of the group.